File: DaxDocumentos/BuscadorDocumentosDax/BuscadorDocumentosEliminar/buscadorDoc.vb

Public Class buscadorDoc

    Public Shared strConxAdcom As String
    Public Shared strConxDaxsys As String

    Public Sub New(strAdc As String, strSys As String)
        strConxAdcom = strAdc
        strConxDaxsys = strSys
    End Sub
    Public Function IniciaBusqueda(ByVal claseDocumentos As String, ByVal codCliente As String, ByVal tipDocAConsultar As String, ByVal IniFec As Date,
            ByRef EsSuc As String, ByRef EsTipDoc As String, ByRef EsNumDoc As Double, ByRef idClaveDocapl As Double, Optional SinArt As Boolean = False _
          , Optional tipDocQueInicia As String = "", Optional ByVal TipoLiquidacion As String = "", Optional ByVal NroLiquidacion As String = "", Optional Tabla As String = "") As Double
        Dim prog As New frmBuscarDoc
        'MsgBox(" " & tipDocOriginal & " - " & codCliente & " - " & tipDocAConsultar & " - " & EsTipDoc)
        With prog
            .claseDoctosPermitidos = claseDocumentos   ' clase de documentos 
            .Codigo = codCliente
            .Solodoc = tipDocAConsultar
            .sinArt = SinArt
            .LiquidacionTip = TipoLiquidacion
            .LiquidacionNum = NroLiquidacion
            ' tipo de documento que llamo a la copiadora
            .DocInicial = tipDocQueInicia
            .tabla = Tabla
            .docSucursal = EsSuc
            If IsDate(IniFec) And IniFec > CDate("0:00:00") Then .InicFec = IniFec Else .InicFec = Now.Date
            .ShowDialog()
            EsSuc = .SucRet
            EsTipDoc = .DocRet
            EsNumDoc = .NumRet
            idClaveDocapl = .idClaveDoc
            IniciaBusqueda = .NumRet
        End With
        prog.Dispose()
    End Function

    Public Function IniciaConsolidacion(ByVal claseDocumentos As String, ByVal codCliente As String, ByVal tipDocAConsultar As String, ByVal IniFec As Date,
        ByRef EsSuc As String, ByRef EsTipDoc As String, ByRef EsNumDoc As Double, ByRef idClaveDocapl As Double, Optional SinArt As Boolean = False _
      , Optional tipDocQueInicia As String = "", Optional ByVal TipoLiquidacion As String = "", Optional ByVal NroLiquidacion As String = "", Optional Tabla As String = "") As String
        Dim prog As New frmBuscarDoc
        'MsgBox(" " & tipDocOriginal & " - " & codCliente & " - " & tipDocAConsultar & " - " & EsTipDoc)
        With prog
            .claseDoctosPermitidos = ""   ' clase de documentos 
            .Codigo = codCliente
            .Solodoc = ""
            .sinArt = SinArt
            .LiquidacionTip = TipoLiquidacion
            .LiquidacionNum = NroLiquidacion
            .DocInicial = claseDocumentos
            .tabla = Tabla
            .docSucursal = EsSuc
            If IsDate(IniFec) And IniFec > CDate("0:00:00") Then .InicFec = IniFec Else .InicFec = Now.Date
            prog.estaConsolidando = True
            .ShowDialog()
            EsSuc = .SucRet
            EsTipDoc = .DocRet
            EsNumDoc = .NumRet
            idClaveDocapl = .idClaveDoc
            IniciaConsolidacion = .DocRet
            Return prog.Lista
        End With
        prog.Dispose()
    End Function
End Class

File: DaxDocumentos/BuscadorDocumentosDax/BuscadorDocumentosEliminar/buscadorDoc.cs
using System;

namespace BuscadorDocumentos
{
    public class buscadorDoc
    {
        public static string strConxAdcom;
        public static string strConxDaxsys;

        public buscadorDoc(string strAdc, string strSys)
        {
            strConxAdcom = strAdc;
            strConxDaxsys = strSys;
        }

        public double IniciaBusqueda(string claseDocumentos, string codCliente, string tipDocAConsultar, DateTime IniFec, ref string EsSuc, ref string EsTipDoc, ref double EsNumDoc, ref double idClaveDocapl, bool SinArt = false, string tipDocQueInicia = "", string TipoLiquidacion = "", string NroLiquidacion = "", string Tabla = "")
        {
            double IniciaBusquedaRet = default;
            var prog = new frmBuscarDoc();
            // MsgBox(" " & tipDocOriginal & " - " & codCliente & " - " & tipDocAConsultar & " - " & EsTipDoc)
            prog.claseDoctosPermitidos = claseDocumentos;   // clase de documentos 
            prog.Codigo = codCliente;
            prog.Solodoc = tipDocAConsultar;
            prog.sinArt = SinArt;
            prog.LiquidacionTip = TipoLiquidacion;
            prog.LiquidacionNum = NroLiquidacion;
            // tipo de documento que llamo a la copiadora
            prog.DocInicial = tipDocQueInicia;
            prog.tabla = Tabla;
            prog.docSucursal = EsSuc;
            if ( IniFec > new DateTime(1900,1,1))
                prog.InicFec = IniFec;
            else
                prog.InicFec = DateTime.Now;
            prog.ShowDialog();
            EsSuc = prog.SucRet;
            EsTipDoc = prog.DocRet;
            EsNumDoc = prog.NumRet;
            idClaveDocapl = prog.idClaveDoc;
            IniciaBusquedaRet = prog.NumRet;
            prog.Dispose();
            return IniciaBusquedaRet;
        }

        public string IniciaConsolidacion(string claseDocumentos, string codCliente, string tipDocAConsultar, DateTime IniFec, ref string EsSuc, ref string EsTipDoc, ref double EsNumDoc, ref double idClaveDocapl, bool SinArt = false, string tipDocQueInicia = "", string TipoLiquidacion = "", string NroLiquidacion = "", string Tabla = "")
        {
            string IniciaConsolidacionRet = default;
            var prog = new frmBuscarDoc();
            // MsgBox(" " & tipDocOriginal & " - " & codCliente & " - " & tipDocAConsultar & " - " & EsTipDoc)
            prog.claseDoctosPermitidos = "";   // clase de documentos 
            prog.Codigo = codCliente;
            prog.Solodoc = "";
            prog.sinArt = SinArt;
            prog.LiquidacionTip = TipoLiquidacion;
            prog.LiquidacionNum = NroLiquidacion;
            prog.DocInicial = claseDocumentos;
            prog.tabla = Tabla;
            prog.docSucursal = EsSuc;
            if (IniFec > new DateTime(1900, 1, 1))
                prog.InicFec = IniFec;
            else
                prog.InicFec = DateTime.Now;
            prog.estaConsolidando = true;
            prog.ShowDialog();
            EsSuc = prog.SucRet;
            EsTipDoc = prog.DocRet;
            EsNumDoc = prog.NumRet;
            idClaveDocapl = prog.idClaveDoc;
            IniciaConsolidacionRet = prog.DocRet;
            return prog.Lista;
        }
    }
}